corehq/apps/app_manager/tests/test_suite_inline_search.py

```diff
@@ -380,7 +376,10 @@ def test_inline_search_with_parent_select(self):
             <form>xmlns1.0</form>
             <post url="http://localhost:8000/a/test_domain/phone/claim-case/"
                 relevant="count(instance('casedb')/casedb/case[@case_id=instance('commcaresession')/session/data/case_id]) = 0">
-              <data key="case_id" ref="instance('commcaresession')/session/data/case_id"/>
+              <data exclude="count(instance('casedb')/casedb/case[@case_id=instance('commcaresession')/session/data/case_id]) != 0"
+              key="case_id" ref="instance('commcaresession')/session/data/case_id"/>
+              <data exclude="count(instance('casedb')/casedb/case[@case_id=instance('commcaresession')/session/data/case_id_case]) != 0"
+              key="case_id" ref="instance('commcaresession')/session/data/case_id_case"/>
```

Comment:
If we're including the `exclude` attribute on each case ID data element then we must remove it from the `post` element. Alternatively `post.relevant` must be updated to do both relevancy checks.

Reply:
> If we're including the exclude attribute on each case ID data element then we must remove it from the post element.

Are you saying the `exclude` attribute needs to be removed from each `data` element  in the `post` element?

> both relevancy checks

do you mean `post.relevant` should be:
`relevant=".../data/case_id]) = 0 and ...//data/case_id_case]) = 0` for each case ID data element?

I'm a bit confused, maybe the answer to my question [here](https://github.com/dimagi/commcare-hq/pull/33738#discussion_r1416202776) will also clarify this

Reply:
As it stands now, the `post.relevant` expression only checks if the case from this module doesn't exists. But if the case from the parent module doesn't exist and the case from this module does then the request won't fire.

I think the ideal for this use case would be remove the `relevant` on the post and only have the `data.exclude` expressions. We'd then need to update the Formplayer logic to only do the post request if there is at least one data element that hasn't been excluded.

@shubham1g5 what do you think?

Reply:
We should be able to set `post.relevant` to `case_id != ''` to achieve this without FP changes.

Reply:
@snopoke What do you think about Shubham's suggestion?

Reply:
that seems great

Reply:
Thanks and just noting a couple findings.

1. From Shubham's recommendation - I checked that [evaluatedParams](https://github.com/dimagi/commcare-core/blob/fe5114b44ebe4128205a903be9d69d3d066baf23/src/main/java/org/commcare/suite/model/PostRequest.java#L73C1-L74C1) contains the `case_ids` for all the data_elements(if not excluded)
2. There was a bug where `$case_id!=''` was evaluating as `true` even if all data elements were excluded. I opened this [PR](https://github.com/dimagi/commcare-core/pull/1390) to address it
3. `entry` elements contain computed datums that contain a new uuid case_id
 `datum id="case_id_new_teacher_0" function="uuid()"/>`.
  As is, a data element is created for `case_id_new_teacher_0`. This would cause `$case_id!=''` to always evaluated to `True`, which causes `error unable to recognize next screen: org.commcare.formplayer.screens.FormplayerSyncScreen@d3d077a:`
So my next step is to omit the respective data element.

Please let me know if any of that doesn't seem correct